### ISSUE TITLE
Windows 10 :term opens git-bash shell outside of vim instead of split buffer

### ISSUE THREAD

**Issue by CrispyDrone** [vim/vim#4923]
**Describe the bug**
I tried replacing the terminal from `cmd.exe` to `git-bash.exe`. Now, when using `:term` it will behave similar to `:sh` although different. A window outside of vim is opened, however a non-responsive and uninitialized split buffer is also created with the cursor locked in the top left corner.
It works fine for `cmd.exe`. In case I set shell to `bash.exe` ($PROGRAMW6432/Git/usr/bin/bash.exe), no commands are found and the terminal shows a message "bash: cut: command not found" when starting. When executing the steps below however, commands are found but the terminal behaves in a very buggy manner constantly overwriting previous text with new text etc.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. `gvim -N -u .vimrc -U NONE -i NONE` where `.vimrc` contains a single line `set shell=$PROGRAMW6432/Git/git-bash.exe`
2. execute `:term`

**Expected behavior**
A new split buffer is opened with a git-bash terminal window that can execute commands such as ls, grep,...

**Screenshots**
![term-01](https://user-images.githubusercontent.com/37639058/64737249-aceb8f80-d4ec-11e9-8131-9dfd837d1086.png)
![term-02](https://user-images.githubusercontent.com/37639058/64737255-af4de980-d4ec-11e9-9e9d-e3598ec7aed3.png)
![term-03](https://user-images.githubusercontent.com/37639058/64737256-b117ad00-d4ec-11e9-9c8c-bb91dd4e9de8.png)

**Environment (please complete the following information):**
 - Vim version output:

```
VIM - Vi IMproved 8.1 (2018 May 18, compiled May 18 2018 18:36:07)
MS-Windows 32-bit GUI version with OLE support
Included patches: 1
Compiled by mool@tororo
Huge version with GUI.  Features included (+) or not (-):
+acl                +channel            +cryptv             -ebcdic             +folding            +lambda             +mksession          +ole                +python3/dyn        +statusline         -tgetent            +visual             -xfontset
+arabic             +cindent            +cscope             +emacs_tags         -footer             +langmap            +modify_fname       +packages           +quickfix           -sun_workshop       -termresponse       +visualextra        -xim
+autocmd            +clientserver       +cursorbind         +eval               +gettext/dyn        +libcall            +mouse              +path_extra         +reltime            +syntax             +textobjects        +viminfo            +xpm_w32
+autoservername     +clipboard          +cursorshape        +ex_extra           -hangul_input       +linebreak          +mouseshape         +perl/dyn           +rightleft          +tag_binary         +timers             +vreplace           -xterm_save
+balloon_eval       +cmdline_compl      +dialog_con_gui     +extra_search       +iconv/dyn          +lispindent         +multi_byte_ime/dyn +persistent_undo    +ruby/dyn           +tag_old_static     +title              -vtp                
-balloon_eval_term  +cmdline_hist       +diff               +farsi              +insert_expand      +listcmds           +multi_lang         -postscript         +scrollbind         -tag_any_white      +toolbar            +wildignore         
+browse             +cmdline_info       +digraphs           +file_in_path       +job                +localmap           +mzscheme/dyn       +printer            +signs              +tcl/dyn            +user_commands      +wildmenu           
++builtin_terms     +comments           +directx            +find_in_path       +jumplist           +lua/dyn            +netbeans_intg      +profile            +smartindent        -termguicolors      +vertsplit          +windows            
+byte_offset        +conceal            -dnd                +float              +keymap             +menu               +num64              +python/dyn         +startuptime        +terminal           +virtualedit        +writebackup        
   system vimrc file: "$VIM\vimrc"
     user vimrc file: "$HOME\_vimrc"
 2nd user vimrc file: "$HOME\vimfiles\vimrc"
 3rd user vimrc file: "$VIM\_vimrc"
      user exrc file: "$HOME\_exrc"
  2nd user exrc file: "$VIM\_exrc"
  system gvimrc file: "$VIM\gvimrc"
    user gvimrc file: "$HOME\_gvimrc"
2nd user gvimrc file: "$HOME\vimfiles\gvimrc"
3rd user gvimrc file: "$VIM\_gvimrc"
       defaults file: "$VIMRUNTIME\defaults.vim"
    system menu file: "$VIMRUNTIME\menu.vim"
Compilation: cl -c /W3 /nologo  -I. -Iproto -DHAVE_PATHDEF -DWIN32  -DFEAT_CSCOPE -DFEAT_TERMINAL -DFEAT_NETBEANS_INTG -DFEAT_JOB_CHANNEL   -DFEAT_XPM_W32   -DWINVER=0x0501 -D_WIN32_WINNT=0x0501 /MP -DHAVE_STDINT_H /Ox /GL -DNDEBUG /arch:IA32 /Zl /MT -DFEAT_OLE -DFEAT_MBYTE_IME -DDYNAMIC_IME -DFEAT_GUI_W32 -DFEAT_DIRECTX -DDYNAMIC_DIRECTX -DFEAT_DIRECTX_COLOR_EMOJI -DDYNAMIC_ICONV -DDYNAMIC_GETTEXT -DFEAT_TCL -DDYNAMIC_TCL -DDYNAMIC_TCL_DLL=\"tcl86t.dll\" -DDYNAMIC_TCL_VER=\"8.6\" -DFEAT_LUA -DDYNAMIC_LUA -DDYNAMIC_LUA_DLL=\"lua53.dll\" -DFEAT_PYTHON -DDYNAMIC_PYTHON -DDYNAMIC_PYTHON_DLL=\"python27.dll\" -DFEAT_PYTHON3 -DDYNAMIC_PYTHON3 -DDYNAMIC_PYTHON3_DLL=\"python36.dll\" -DFEAT_MZSCHEME -I "E:\Racket\include" -DMZ_PRECISE_GC -DDYNAMIC_MZSCHEME -DDYNAMIC_MZSCH_DLL=\"libracket3m_a36fs8.dll\" -DDYNAMIC_MZGC_DLL=\"libracket3m_a36fs8.dll\" -DFEAT_PERL -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DDYNAMIC_PERL -DDYNAMIC_PERL_DLL=\"perl524.dll\" -DFEAT_RUBY -DDYNAMIC_RUBY -DDYNAMIC_RUBY_VER=24 -DDYNAMIC_RUBY_DLL=\"msvcrt-ruby240.dll\" -DFEAT_HUGE /Fd.\ObjGXOULYHTRZi386/ /Zi
Linking: link  /nologo /subsystem:windows,5.01 /opt:ref /LTCG:STATUS oldnames.lib kernel32.lib advapi32.lib shell32.lib gdi32.lib  comdlg32.lib ole32.lib uuid.lib /machine:i386 gdi32.lib version.lib   winspool.lib comctl32.lib advapi32.lib shell32.lib  /machine:i386  libcmt.lib oleaut32.lib user32.lib  /nodefaultlib:lua53.lib  /STACK:8388608  /nodefaultlib:python27.lib /nodefaultlib:python36.lib   "E:\ActiveTcl\lib\tclstub86.lib" WSock32.lib xpm\x86\lib-vc14\libXpm.lib /PDB:gvim.pdb -debug
```

 - OS: Windows 10 Home Version 1903 OS build 18362.295
 - Terminal: GUI 
- Git: version 2.22.0.windows.1
- Mintty: mintty 3.0.1 (x86_64-pc-msys)
  Type: xterm

**Comment by k-takata** [comment]
`git-bash.exe` is not a shell itself. It is a launcher which launches bash inside mintty.
Try `set shell=$PROGRAMW6432/Git/bin/bash.exe` instead.

**Comment by CrispyDrone** [comment]
```
set shell=$PROGRAMW6432/Git/bin/bash.exe
```
Behaves as described in my post. However it is buggy with  `clear` only resulting in the cursor jumping to the top of the screen and subsequent commands overwriting previous text. 

I've managed to discover another issue related to how gvim is started. When I start gvim from my task bar it directly invokes "C:\Program Files (x86)\Vim\vim81\gvim.exe".

![image](https://user-images.githubusercontent.com/37639058/64806837-4b7efb80-d594-11e9-8f0b-54aa9dec7f4b.png)

However when I start gvim from git-bash.exe it uses `C:\Windows\gvim.bat`. In this latter commands such as `ls`, `grep` all function properly! However the `clear` command still results in the exact same thing as I described earlier. 

![image](https://user-images.githubusercontent.com/37639058/64806902-694c6080-d594-11e9-91a6-71aff5e0383f.png)

![image](https://user-images.githubusercontent.com/37639058/64807013-9b5dc280-d594-11e9-8857-266eb31c4b21.png)

Now when I change the way the shortcut starts gvim to use `C:\Windows\gvim.bat`, it still doesn't work. None of the commands are found.

Any idea what this might be and how to solve it?

**Update**: Actually sorry about all that... I was using `$PROGRAMDATAW6432/Git/usr/bin/bash` instead of `$PROGRAMDATAW6432/Git/bin/bash`... Using the latter is working as expected. Thank you. Any idea why the former doesn't work?
